lib/pbench/test/functional/server/test_put.py

```diff
@@ -142,6 +199,40 @@ def test_index_all(self, server_client: PbenchServerClient, login_user):
         ), f"Timed out after {now - start} seconds; unindexed datasets: " + ", ".join(
             d.metadata["server.tarball-path"] for d in not_indexed
         )
-        assert count == len(
-            self.tarballs
-        ), f"Didn't find all expected datasets, found {count} of {len(self.tarballs)}"
+        assert (
+            len(tarball_names) == count
+        ), f"Didn't find all expected datasets, found {count} of {len(tarball_names)}"
+
+    @staticmethod
+    def test_delete_all(server_client: PbenchServerClient, login_user):
+        """Verify we can delete each previously uploaded dataset.
+
+        Requires that test_upload_all has been run successfully.
+        """
+        print(" ... reporting behaviors ...")
+
+        datasets = server_client.get_list()
+        datasets_hash = {}
+        try:
+            for dataset in datasets:
+                datasets_hash[f"{dataset.name}.tar.xz"] = dataset.resource_id
+        except HTTPError as exc:
+            pytest.fail(
+                f"Unexpected HTTP error, url = {exc.response.url}, status code"
+                f" = {exc.response.status_code}, text = {exc.response.text!r}"
+            )
+        for t in TARBALL_DIR.glob("*.tar.xz"):
+            resource_id = datasets_hash.get(t.name)
+            assert resource_id, f"Expected to find tar ball {t.name} to delete"
+            response = server_client.post(
+                api=API.DATASETS_DELETE,
+                uri_params={"dataset": resource_id},
+                raise_error=False,
+            )
+            assert response.ok, f"{response.text}"
+            print(f"Deleted {t.name}")
+            response = server_client.head(
```

Comment:
This fails with

```
"HEAD /api/v1/datasets/metadata/22a4bc5748b920c6ce271eb68f08d91c HTTP/1.1" 405 0
```
I guess this was what was being discussed at the backlog grooming: `HEAD` is not allowed?

Reply:
This is a `Draft` PR and this change is not correct.

Reply:
So given #3146, this should be closed?

Reply:
Once it is rebased, there is other work that this relies on from other PRs which may prove useful.

I am not going to do anything with this PR for the time being.